src/ert/ensemble_evaluator/snapshot.py

```diff
@@ -240,179 +208,209 @@ def _merge(self, other: "PartialSnapshot") -> "PartialSnapshot":
             self._forward_model_states[forward_model_id].update(other_fm_data)
         return self
 
-    def from_cloudevent(self, event: CloudEvent) -> "PartialSnapshot":
+    def update_from_cloudevent_and_generate_update_snapshot(
```

Comment:
It is not clear to me from the docstring if `self` is updated also. Since we actually return something it seems not. And then we should probably write `..updated_snapshot` and not `update_snapshot`.

Reply:
This function is weird, because it updates itself with the event (inplace?), and also returns the update_snapshot only containing the change. The update snapshot (way smaller than actual snapshot) is the one we send to the GUI to avoid resource starvation.